File: app/src/main/java/com/spacex/launches/spacexrocketlaunches/rocketList/RocketUIMDecorator.java
package com.spacex.launches.spacexrocketlaunches.rocketList;

import android.content.Context;
import android.graphics.drawable.Drawable;
import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import com.spacex.launches.spacexrocketlaunches.R;

public class RocketUIMDecorator {

  private final Context context;
  private RocketUIM rocketUIM;

  RocketUIMDecorator(@NonNull final Context context) {
    this.context = context;
  }

  public RocketUIMDecorator bind(@NonNull final RocketUIM rocketUIM) {
    this.rocketUIM = rocketUIM;
    return this;
  }

  public Drawable getImage() {
    return ContextCompat.getDrawable(context, rocketUIM.getImageResId());
  }

  public String getName() {
    return rocketUIM.getName();
  }

  public String getDetails() {
    return context.getString(
        R.string.rocket_list_details_format, rocketUIM.getEnginesCount(), rocketUIM.getCountry());
  }
}

File: app/src/main/kotlin/com/spacex/ui/rocketList/RocketUIMDecorator.kt
package com.spacex.ui.rocketList

import android.content.Context
import android.graphics.drawable.Drawable
import androidx.core.content.ContextCompat
import com.spacex.ui.R

internal class RocketUIMDecorator(private val context: Context) {
    private var rocketUIM: RocketUIM? = null

    fun bind(rocketUIM: RocketUIM): RocketUIMDecorator {
        this.rocketUIM = rocketUIM
        return this
    }

    val image: Drawable?
        get() = ContextCompat.getDrawable(context, rocketUIM!!.imageResId)

    val name: String
        get() = rocketUIM!!.name

    val details: String
        get() = context.getString(
                R.string.rocket_list_details_format, rocketUIM!!.enginesCount, rocketUIM!!.country)
}
